opencti-platform/opencti-graphql/src/database/engine.js

```diff
@@ -3595,15 +3694,86 @@ const prepareIndexing = async (elements) => {
   }
   return preparedElements;
 };
+// Creates a copy of a live element in the draft index with the current draft context
+const copyLiveElementToDraft = async (context, user, element) => {
+  const draftContext = getDraftContext(context, user);
+  if (!draftContext || element._index.includes(INDEX_DRAFT)) return element;
+
+  const updatedElement = structuredClone(element);
+  const newId = generateInternalId();
+  const setDraftData = `ctx._id="${newId}";ctx._source.draft_ids=['${draftContext}'];`;
+  await elReindexElements(context, user, [element.internal_id], element._index, INDEX_DRAFT, setDraftData);
+  updatedElement._id = newId;
+  updatedElement._index = INDEX_DRAFT;
+
+  // Add draftId to live element draftsIds
+  const addDraftIdScript = {
+    script: {
+      source: `
+        if (ctx._source.containsKey('draft_ids')) 
+          {ctx._source['draft_ids'].add('${draftContext}');} 
+        else 
+          {ctx._source.draft_ids = ['${draftContext}']}
+      `
+    }
+  };
+  await elUpdate(element._index, element.internal_id, addDraftIdScript);
+
+  return updatedElement;
+};
+// Gets the version of the element in current draft context if it exists
+// If it doesn't exist, creates a copy of live element to draft context then returns it
+const getElementDraftVersion = async (context, user, element) => {
```

Comment:
see if we can find another naming since this method gets and creates if it doesn't exist